src/network/networkd-route.c

```diff
@@ -436,6 +437,10 @@ int route_configure(Route *route, Link *link,
         assert(link->manager);
         assert(link->manager->rtnl);
         assert(link->ifindex > 0);
+
+        if (route->family == AF_UNSPEC)
+                route->family = AF_INET;
```

Comment:
This comes from the same 
```
ip route add default dev
```
If we don't set anything it's been set to ipv4. In this case we don't give a gateway.

```
Gateway=0.0.0.0 works but not `::`
￼```